internal/ioutil/ioutil.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ioutil
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+// Ingest writes content into a temporary ingest file with the file name format
+// "oras_credstore_temp_{randomString}".
+func Ingest(dir string, content io.Reader) (path string, ingestErr error) {
+	tempFile, err := os.CreateTemp(dir, "oras_credstore_temp_*") // mode: 0600
```

Comment:
Where is the reference for the mode `0600`?

Reply:
Let's manually chmod.